packages/samples/headless-react/src/pages/DependentFacet.tsx

```diff
@@ -0,0 +1,86 @@
+import {
+  buildFacet,
+  buildSearchEngine,
+  Facet as HeadlessFacet,
+  getSampleSearchEngineConfiguration,
+  loadFacetSetActions,
+  SearchEngine,
+} from '@coveo/headless';
+import {Component} from 'react';
+import {Facet} from '../components/facet/facet.fn';
+import {AppContext} from '../context/engine';
+
+export class DependentFacet extends Component<
```

Comment:
Would it be possible to demo the feature using a `function` component?
I think React devs are increasingly migrating their code-bases to the functional style, so aligning the samples could be helpful,